pkg/pillar/types/global_test.go

```diff
@@ -197,63 +278,81 @@ func TestParseGlobalItem(t *testing.T) {
 }
 func TestParseAgentItem(t *testing.T) {
 	// log.SetLevel(log.DebugLevel)
-	specMap := ConfigItemSpecMap{}
-	specMap.AgentSettings = make(map[AgentSettingKey]ConfigItemSpec)
-	specMap.AgentSettings[LogLevel] = ConfigItemSpec{
-		ItemType:        ConfigItemTypeString,
-		StringValidator: parseLevel,
-	}
-	specMap.AgentSettings[RemoteLogLevel] = ConfigItemSpec{
-		ItemType:        ConfigItemTypeString,
-		StringValidator: parseLevel,
-	}
+	specMap := NewConfigItemSpecMap()
 	globalConfig := ConfigItemValueMap{}
 	globalConfig.AgentSettings = make(map[string]map[AgentSettingKey]ConfigItemValue)
 	testMatrix := map[string]struct {
-		item     config.ConfigItem
-		logLevel AgentSettingKey
+		item        config.ConfigItem
+		logLevel    AgentSettingKey
+		expectError bool
 	}{
-		"Agent Setting": {
+		"Agent Setting LogLevel New": {
 			logLevel: LogLevel,
 			item: config.ConfigItem{
 				Key:   "agent.zedagent.debug.loglevel",
 				Value: "info",
 			},
```

Comment:
Shouldn’t we test with something different than the default value?

Reply:
Good point.. Done..
Added more detailed test cases..